src/integrals/integrals.h

```diff
@@ -224,19 +244,30 @@ class ForteIntegrals {
     virtual void set_tei(size_t p, size_t q, size_t r, size_t s, double value, bool alpha1,
                          bool alpha2) = 0;
 
-    /// Update the integrals with a new set of MO coefficients
-    virtual void retransform_integrals();
+    /// Rotate the MO coefficients, update psi::Wavefunction, and re-transform integrals
+    /// @param Ua the alpha unitary transformation matrix
+    /// @param Ub the alpha unitary transformation matrix
+    void rotate_orbitals(std::shared_ptr<psi::Matrix> Ua, std::shared_ptr<psi::Matrix> Ub);
+
+    /// Copy these MO coeffs to class variables, update psi::Wavefunction, and re-transform
+    /// integrals
+    /// @param Ca the alpha MO coefficients
+    /// @param Cb the betaa MO coefficients
+    void update_orbitals(std::shared_ptr<psi::Matrix> Ca, std::shared_ptr<psi::Matrix> Cb);
+
     /// Expert Option: just try and use three_integral
     virtual double** three_integral_pointer() = 0;
```

Comment:
Remove `virtual double** three_integral_pointer()`. It is never used and we usually do not use type `double**`.

Reply:
This is called in three_dsrg_mrpt2.cc

Reply:
Good catch!